src/MyMarketManager.Processing/ImageVectorizationHandler.cs

```diff
@@ -0,0 +1,80 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using MyMarketManager.Data;
+
+namespace MyMarketManager.Processing;
+
+/// <summary>
+/// Handler that fetches product photos without vector embeddings and generates embeddings.
+/// Can be registered multiple times with different configurations for different sources
+/// (e.g., product photos, delivery photos, etc.).
+/// </summary>
+public class ImageVectorizationHandler : IWorkItemHandler<ImageVectorizationWorkItem>
```

Comment:
I think we are going to have several handlers so I am thinking we should make this a base class and then have a ProductPhotoIVH that derives from this to load prod photo and process specifically.